rye/src/pyproject.rs

```diff
@@ -919,8 +924,12 @@ fn resolve_lower_bound_python_version(doc: &Document) -> Option<PythonVersionReq
         })
 }
 
-pub fn find_project_root() -> Option<PathBuf> {
-    let mut here = env::current_dir().ok()?;
+pub fn find_project_root_from(path: PathBuf) -> Option<PathBuf> {
+    let mut here = if path.is_relative() {
+        path.canonicalize().ok()?
```

Comment:
We don't really like canonicalize and would like to use the unstable std function for absolute path.

Reply:
One rather easy way to absolutize a path without canonicalizing is `std::current_dir()?.join(path)`. Not sure if that's a better option here.

Reply:
I found that in the repo (after writing the comment) but I also tested it out on `--pyproject-from ../../../`  which kind of has a bad failure mode, because it will descend instead of ascend while looking for a pyproject.toml file. Once I'd seen that, can't use it :slightly_smiling_face: